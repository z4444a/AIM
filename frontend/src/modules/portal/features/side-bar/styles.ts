import { Theme } from '@material-ui/core/styles/createMuiTheme';
import createStyles from '@material-ui/core/styles/createStyles';
import { WithStyles } from '@material-ui/core/styles/withStyles';
import { StyleRules } from '@material-ui/core/styles';

const drawerWidth = 200;

export const styles = (theme: Theme): StyleRules =>
  createStyles({
    toolbar: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      padding: '0 8px',
      ...theme.mixins.toolbar,
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0,
      whiteSpace: 'nowrap',
    },
    drawerOpen: {
      width: drawerWidth,
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
      }),
    },
    drawerClose: {
      transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
      }),
      overflowX: 'hidden',
      width: theme.spacing.unit * 7 + 1,
      [theme.breakpoints.up('sm')]: {
        width: theme.spacing.unit * 9 + 1,
      },
    },
    drawerPaper: {
      width: drawerWidth,
    },
    link: {
      textDecoration: 'none',
    },
  });

export type Style = WithStyles<
  'drawer' | 'drawerOpen' | 'drawerClose' | 'drawerPaper' | 'toolbar' | 'link'
>;
