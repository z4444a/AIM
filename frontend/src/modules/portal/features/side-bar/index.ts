import { connect } from 'react-redux';
import withStyles from '@material-ui/core/styles/withStyles';
import { styles } from './styles';
import { RootState } from 'typesafe-actions';
import { SideBar } from './side-bar';

export const mapStateToProps = (state: RootState) => ({
  sideNavIsOpen: state.portal.sideNavIsOpen,
});

export default connect(mapStateToProps)(withStyles(styles)(SideBar));
