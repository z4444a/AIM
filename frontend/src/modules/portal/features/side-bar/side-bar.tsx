import React, { ReactNode } from 'react';
import { Style } from './styles';
import { mapStateToProps } from './index';
import Drawer from '@material-ui/core/Drawer';
import MainMenu from '../main-menu/index';

export type InternalProps = Style & ReturnType<typeof mapStateToProps>;

export class SideBar extends React.PureComponent<InternalProps> {
  public render(): ReactNode {
    const { classes, sideNavIsOpen } = this.props;
    const stateClass = sideNavIsOpen ? classes.drawerOpen : classes.drawerClose;
    return (
      <Drawer
        className={[classes.drawer, stateClass].join(' ')}
        variant="permanent"
        classes={{
          paper: stateClass,
        }}
        open={sideNavIsOpen}
      >
        <div className={classes.toolbar} />
        <MainMenu />
      </Drawer>
    );
  }
}
