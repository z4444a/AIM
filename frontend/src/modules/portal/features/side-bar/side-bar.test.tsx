import { shallow, ShallowWrapper } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { InternalProps, SideBar } from './side-bar';

describe('SideBar', () => {
  let wrapper: ShallowWrapper;
  const props: InternalProps = {
    classes: {
      drawer: 'drawer',
      drawerOpen: 'drawerOpen',
      drawerClose: 'drawerClose',
      drawerPaper: 'drawerPaper',
      toolbar: 'toolbar',
      link: 'link',
    },
    sideNavIsOpen: true,
  };
  beforeEach(() => {
    wrapper = shallow(<SideBar {...props} />);
  });

  it('should render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
