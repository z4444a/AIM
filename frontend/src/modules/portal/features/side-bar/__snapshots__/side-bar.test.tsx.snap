// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar should render correctly 1`] = `
<WithStyles(Drawer)
  className="drawer drawerOpen"
  classes={
    Object {
      "paper": "drawerOpen",
    }
  }
  open={true}
  variant="permanent"
>
  <div
    className="toolbar"
  />
  <Connect(withI18nextTranslation(MainMenu)) />
</WithStyles(Drawer)>
`;
